spanner/cloud-client/src/main/java/com/example/spanner/LoadCsvExample.java

```diff
@@ -0,0 +1,274 @@
+import org.apache.commons.cli.Options;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+
+/** Sample showing how to load CSV file data into Spanner */
+public class LoadCsvExample {
+  public static final String EXCEL = "EXCEL";
+  public static final String POSTGRESQL_CSV = "POSTGRESQL_CSV";
+  public static final String POSTGRESQL_TEXT = "POSTGRESQL_TEXT}";
+
+  enum SpannerDataTypes {
+    STRING,
+    INT64,
+    FLOAT64,
+    BYTE,
+    BOOL,
+    DATE,
+    TIMESTAMP
+  }
+
+  public static Boolean hasHeader = false;
+  public static Connection connection;
+  public static Map<String, SpannerDataTypes> tableColumns = new LinkedHashMap<>();
+
+  /** Return the data type of the column type **/
+  public static SpannerDataTypes parseSpannerDataType(String columnType) {
+    if (columnType.matches("STRING(?:\\((?:MAX|[0-9]+)\\))?")) {
+      return SpannerDataTypes.STRING;
+    } else if (columnType.equalsIgnoreCase("INT64")) {
+      return SpannerDataTypes.INT64;
+    } else if (columnType.equalsIgnoreCase("FLOAT64")) {
+      return SpannerDataTypes.FLOAT64;
+    } else if (columnType.equalsIgnoreCase("BYTE")) {
+      return SpannerDataTypes.BYTE;
+    } else if (columnType.equalsIgnoreCase("BOOL")) {
+      return SpannerDataTypes.BOOL;
+    } else if (columnType.equalsIgnoreCase("DATE")) {
+      return SpannerDataTypes.DATE;
+    } else if (columnType.equalsIgnoreCase("TIMESTAMP")) {
+      return SpannerDataTypes.TIMESTAMP;
+    } else {
+      throw new IllegalArgumentException(
+          "Unrecognized or unsupported column data type: " + columnType);
+    }
+  }
+
+  /** Query database for column names and types in the table **/
+  public static void parseTableColumns(String tableName) throws SQLException {
+    ResultSet spannerType = connection.createStatement()
+        .executeQuery("SELECT column_name, spanner_type FROM information_schema.columns "
+            + "WHERE table_name = \"" + tableName + "\" ORDER BY ordinal_position");
+    while (spannerType.next()) {
+      String columnName = spannerType.getString("column_name");
+      SpannerDataTypes type = parseSpannerDataType(spannerType.getString("spanner_type"));
+      tableColumns.put(columnName, type);
+    }
+  }
+
+  /** Check that CSV file headers exist as a table column name **/
+  public static boolean isValidHeader(CSVParser parser) {
+    List<String> csvHeaders = parser.getHeaderNames();
+    for (String csvHeader : csvHeaders) {
+      if (!tableColumns.containsKey(csvHeader)) {
+        throw new IllegalArgumentException(
+            "File header " + csvHeader + " does not match any database table column name.");
+      }
+    }
+    return true;
+  }
+
+  /** Initialize CSV Parser format based on user specified option flags **/
+  public static CSVFormat setFormat(CommandLine cmd) {
+    CSVFormat parseFormat;
+    // Set file format type
+    if (cmd.hasOption("f")) {
```

Comment:
I believe these were just the formats specified in the initial design doc - EXCEL being assumed to be one of the more popular CSV formats and otherwise PostgreSQL CSV / TEXT being used here because they will be reused again in the main PGAdapter project as well.